src/providers/ad/ad_gpo.c

```diff
@@ -864,19 +864,20 @@ ad_gpo_filter_gpos_by_dacl(TALLOC_CTX *mem_ctx,
 
         access_allowed = false;
         candidate_gpo = candidate_gpos[i];
-        sd = candidate_gpo->gpo_sd;
-        dacl = candidate_gpo->gpo_sd->dacl;
```

Comment:
This is not visible from code. There is not even comment that `candidate_gpo->gpo_func_version != 2` is somehow related to the `candidate_gpo->gpo_sd == NULL`.

Maybe it would be better to add additional check afte checking `gpo_func_version` and fail if `candidate_gpo->gpo_sd` is `NULL`